components/org.wso2.carbon.identity.api.server.organization.management/org.wso2.carbon.identity.api.server.organization.management.v1/src/main/java/org/wso2/carbon/identity/api/server/organization/management/v1/service/OrganizationManagementService.java

```diff
@@ -570,8 +569,6 @@ private Organization getOrganizationFromPostRequest(OrganizationPOSTRequest orga
         String parentId = organizationPOSTRequest.getParentId();
         if (StringUtils.isNotBlank(parentId)) {
             organization.getParent().setId(parentId);
-        } else {
-            organization.getParent().setId(SUPER);
```

Comment:
The initial commit was to set the parent ID taken from the carbon context. But it is null as the organziationID is set to null when tenanted URL accesses.

Hence logic was changed to skip setting parent ID if not provided by treating it as an optional attribute. The accessing organization ID is set as the parent ID at the organization management service.